external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaTupleListener.java

```diff
@@ -0,0 +1,48 @@
+package org.apache.storm.kafka.spout;
+
+import java.io.Serializable;
+import java.util.List;
+
+
+/**
+ * The KafkaTupleListener handles state changes of a kafka tuple inside a KafkaSpout.
+ */
+public interface KafkaTupleListener extends Serializable {
```

Comment:
It might be useful to have a prepare method on this interface too, in case implementations need to instantiate something that can't be serialized, e.g. a KafkaProducer. I'm also wondering if it would be useful to be able to emit dead tuples to a different stream as well?